File: server/app/models/db/general_resource/course_response.py
from pydantic import BaseModel
from app.db.enums import AvailableCourse, AvailableDialect, AvailableLanguage


class CourseResponse(BaseModel):
    id: int
    course_name: AvailableCourse
    total_modules: int
    image: str
    text_color: str
    default_dialect: AvailableDialect | None = None
    language: AvailableLanguage

File: client/types/response_models/CourseResponse.ts
import { AvailableCourse, AvailableDialect, AvailableLanguage } from "../enums";

export interface CourseResponse {
    id: number;
    course_name: AvailableCourse;
    total_modules: number;
    image: string;
    text_color: string;
    default_dialect: AvailableDialect | null;
    language: AvailableLanguage;
}
